inst/tinytest/test_read_metadata.R

```diff
@@ -13,3 +13,34 @@ expect_equal(dim(df), c(1, 6))
 
 ## write details to screen
 expect_stdout(zarr_overview(zarr_c, as_data_frame = FALSE))
+
+## error when metadata file is not found (local zarr array)
+zarr_c <- system.file(
+  "extdata",
+  "zarr_examples",
+  "missing-metadata",
+  "int64.zarr",
+  package = "Rarr"
+)
+
+expect_error(
+  zarr_overview(zarr_c),
+  "The array does not have an associated .zarray metadata file"
+)
+
+## error when metadata file is not found (zarr array on s3)
+if (require(mockery)) {
+  # Ensures that .s3_object_exists() just automatically returns FALSE
+  stub(
+    where = Rarr:::read_array_metadata,
+    what = ".s3_object_exists",
+    how = FALSE
+  )
+
+  expect_error(
+    zarr_overview("https://objects.eodc.eu:443/e05ab01a9d56408d82ac32d69a5aae2a:202505-s02msil2a/30/products/cpm_v256/S2B_MSIL2A_20250530T101559_N0511_R065_T32TPT_20250530T130924.zarr/measurements/reflectance/r10m/",
```

Comment:
What about

```suggestion
    zarr_overview("http://s3.example.com/bucket/example.zarr/",
```

?

To keep is short & simple and highlight the error is not bound to a specific set up / server.

Reply:
Yup sounds great! You're right, the actual URL is not relevant.